### ISSUE TITLE
feat: Add EarlyStopping callback with best weights restoration

### ISSUE THREAD

**Issue by Genajoin** [Genajoin/Paraglidable#10]
## Feature Description

Add custom EarlyStopping callback for automatic training stopping.

## Changes

- New `src/pyparaglide/training/early_stopping.py` module
- Custom implementation (not relying on Keras built-in)
- Add `--early-stopping-patience` parameter to CLI

## Features

- Monitors validation loss
- Stops when no improvement for N epochs
- Restores best weights automatically
- Works with both validation strategies (alternating days & Keras split)
- Verbose logging with green/red indicators

## Usage

```bash
# Enable early stopping with 30 epoch patience
pyparaglide train --model cells --epochs 600 --early-stopping-patience 30
```

## Implementation Details

- `min_delta=0.001` minimum change to qualify as improvement
- Logs every 5 epochs of waiting (not every epoch)
- Clear summary when stopping

Part of #XXX (training improvements)
